R/cohensFleissKappa.R

```diff
@@ -0,0 +1,184 @@
+    if (weighted && length(levels(unlist(dataset))) < 3)
+      jaspTable$addFootnote(gettext('If there are only 2 levels, weighted kappa is equal to unweighted kappa.'))
+  }
+  return(jaspTable)
+}
+
+.computeFleissKappaTable <- function(dataset, options, ready) {
+  
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Fleiss' Kappa"))
+  jaspTable$addColumnInfo(name = "ratings", title = gettext("Ratings"), type = "string")
+  jaspTable$addColumnInfo(name = "fKappa", title = gettext("Fleiss' Kappa"), type = "number")
+  jaspTable$position <- 2
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "fleissKappa",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  if (ready) {
+    #calculate Fleiss' Kappa
+    allKappaData <- .fleissKappaMod(dataset, detail = TRUE)
+    overallKappa <- allKappaData$value
+    categoryKappas <- allKappaData$detail[, 1]
+    overallSE <- allKappaData$se
+    categorySE <- allKappaData$se_cat
+    alpha <- 1 - options[["kappaConfidenceIntervalValue"]]
+    
+    # for nominal text data we want the rating text to be displayed:
+    # if the transformation to numeric goes wrong:
+    if (anyNA(as.numeric(as.character(unique(unlist(dataset)))))) {
+      categories <- sort(unique(unlist(dataset)))
+    } else { # if data is ordinal (can be transformed to numeric)
+      categories <- sort(as.numeric(as.character(unique(unlist(dataset)))))
+    }
+    ratings <- c("Overall", as.character(categories))
+    # when the data was read with columns as factors the kappa function
+    # would not order the categories numerically, this is a fix
+    categoryKappas <- categoryKappas[as.character(categories)]
+    
+    tableData <- list("ratings" = ratings,
+                      "fKappa" = c(overallKappa, categoryKappas))
+    if (options[["kappaIntervalOn"]]) {
+      overallCI <- overallKappa + c(-1,1)*qnorm(1 - alpha/2) * overallSE 
+      categoryCIL <- categoryKappas - qnorm(1 - alpha/2) * categorySE
+      categoryCIU <- categoryKappas + qnorm(1 - alpha/2) * categorySE
+      jaspTable$addColumnInfo(name = "CIL", title = gettext("Lower"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      jaspTable$addColumnInfo(name = "CIU", title = gettext("Upper"), type = "number", overtitle = gettextf("%s%% CI for Kappa", formattedCIPercent))
+      tableData[["CIL"]] <- c(overallCI[1], categoryCIL)
+      tableData[["CIU"]] <- c(overallCI[2], categoryCIU)
+      
+      jaspTable$addFootnote(gettext('Confidence intervals are asymptotic.'))
+    }
+    
+    jaspTable$setData(tableData)
+    
+    jaspTable$addFootnote(gettextf('%i subjects/items and %i raters/measurements.', nrow(dataset), ncol(dataset)))
+  }
```

Comment:
You are creating two footnotes here, which looks a bit weird: 
![Screenshot 2021-12-13 at 11 06 47](https://user-images.githubusercontent.com/38500953/145792636-7371ec32-32d5-495f-8482-fde07cd25371.png)
I think it would be better to save each footnote in a variable and then paste them together into one footnote that you can then add. I also wonder if the cohens kappa confidence intervals are also asymptotic. I would think yes, since they are normal theory intervals which usually depend on big numbers